core/services/headreporter/telemetry_reporter.go

```diff
@@ -0,0 +1,53 @@
+package headreporter
+
+import (
+	"context"
+	evmtypes "github.com/smartcontractkit/chainlink/v2/core/chains/evm/types"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/legacyevm"
+	"github.com/smartcontractkit/chainlink/v2/core/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization"
+	"github.com/smartcontractkit/chainlink/v2/core/services/synchronization/telem"
+	"github.com/smartcontractkit/chainlink/v2/core/services/telemetry"
+	"github.com/smartcontractkit/libocr/commontypes"
+	"google.golang.org/protobuf/proto"
+	"math/big"
+)
+
+type (
+	telemetryReporter struct {
+		logger    logger.Logger
+		endpoints map[*big.Int]commontypes.MonitoringEndpoint
+	}
+)
+
+func NewTelemetryReporter(chainContainer legacyevm.LegacyChainContainer, lggr logger.Logger, monitoringEndpointGen telemetry.MonitoringEndpointGenerator) *telemetryReporter {
+	endpoints := make(map[*big.Int]commontypes.MonitoringEndpoint)
```

Comment:
A pointer is not a good key for a map, as any new pointer (even with same value) will have a different place in memory; if it is chainID, it's safe to use uint64

Reply:
done